node/src/accountant/db_access_objects/sent_payable_dao.rs

```diff
@@ -574,166 +552,157 @@ mod tests {
+            gas_price_wei_high_b, \
+            gas_price_wei_low_b, \
+            nonce, \
+            block_hash, \
+            block_number\
+            )
+               VALUES (?1, ?2, ?3, ?4, ?5, ?6, ?7, ?8, ?9, ?10)";
+            let mut stmt = wrapped_conn.prepare(sql).unwrap();
+            stmt.execute(rusqlite::params![
+                "0x1234567890123456789012345678901234567890123456789012345678901234",
+                "0x1234567890123456789012345678901234567890",
+                0,
+                100,
+                1234567890,
+                0,
+                1000000000,
+                1,
+                "0x2345678901234567890123456789012345678901234567890123456789012345",
+                rusqlite::types::Null,
+            ])
             .unwrap();
+        }
 
-        let result = subject.retrieve_txs(Some(ByHash(vec![tx1.hash])));
+        let subject = SentPayableDaoReal::new(wrapped_conn);
 
-        assert_eq!(result, vec![tx1]);
+        // This should panic due to invalid block details
+        let _ = subject.retrieve_txs(None);
     }
 
     #[test]
-    fn change_statuses_works() {
+    fn update_tx_blocks_works() {
         let home_dir =
-            ensure_node_home_directory_exists("sent_payable_dao", "change_statuses_works");
+            ensure_node_home_directory_exists("sent_payable_dao", "update_tx_blocks_works");
         let wrapped_conn = DbInitializerReal::default()
             .initialize(&home_dir, DbInitializationConfig::test_default())
             .unwrap();
         let subject = SentPayableDaoReal::new(wrapped_conn);
-        let tx1 = TxBuilder::default()
-            .hash(H256::from_low_u64_le(1))
-            .status(TxStatus::Pending)
-            .build();
-        let tx2 = TxBuilder::default()
-            .hash(H256::from_low_u64_le(2))
-            .status(TxStatus::Pending)
-            .build();
+        let tx1 = TxBuilder::default().hash(make_tx_hash(1)).build();
+        let tx2 = TxBuilder::default().hash(make_tx_hash(2)).build();
         subject
             .insert_new_records(&vec![tx1.clone(), tx2.clone()])
             .unwrap();
+        let tx_block_1 = TransactionBlock {
+            block_hash: make_block_hash(3),
+            block_number: U64::from(1),
+        };
+        let tx_block_2 = TransactionBlock {
+            block_hash: make_block_hash(4),
+            block_number: U64::from(2),
+        };
         let hash_map = HashMap::from([
-            (tx1.hash, TxStatus::Failed),
-            (
-                tx2.hash,
-                TxStatus::Succeeded(TransactionBlock {
-                    block_hash: H256::from_low_u64_le(3),
-                    block_number: U64::from(1),
-                }),
-            ),
+            (tx1.hash, tx_block_1.clone()),
+            (tx2.hash, tx_block_2.clone()),
         ]);
 
-        let result = subject.change_statuses(&hash_map);
+        let result = subject.update_tx_blocks(&hash_map);
```

Comment:
In this test, I'd maybe prefer to have another call like this before the act. So that you can assert that the records had the block column empty before the test itself starts.